### PR TITLE
Remove `UserPermissionManageableApplicationPolicy`

### PR DESCRIPTION
**Pull request by yndajas**
This policy is a little complex, but is being used in quite niche ways. Simplifying or removing it will make the access and permissions logic a little less nested/complex and therefore easier to change

This opts for removing it, given it's not used as consistently as might be expected (based on its name)

---

This application is owned by the publishing platform team. Please let us know in #govuk-publishing-platform when you raise any PRs.

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: test/policies/supported_permission_policy_scope_test.rb [hunk 1]
```diff
@@ -23,6 +25,9 @@ class SupportedPermissionPolicyScopeTest < ActiveSupport::TestCase
 
     @app_four_pat_permission = create(:non_delegatable_supported_permission, application: @app_three, name: "pat")
     @app_four_sat_permission = create(:delegatable_supported_permission, application: @app_three, name: "sat")
+
+    @api_only_app_nat_permission = create(:non_delegatable_supported_permission, application: @api_only_app, name: "nat")
```
Comment (Gweaton) [inline test/policies/supported_permission_policy_scope_test.rb:29]: These names 😄

File: test/policies/supported_permission_policy_scope_test.rb [hunk 2]
```diff
@@ -31,133 +31,72 @@ class SupportedPermissionPolicyScopeTest < ActiveSupport::TestCase
...
-        assert_includes @resolved_scope, @app_one_cat_permission
-        assert_includes @resolved_scope, @app_one_hat_permission
-      end
-
-      should "not contain any permissions for apps with non-delegatbale signin permission the organisation admin has access to" do
-        assert_not_includes @resolved_scope, @app_two_signin_permission
-        assert_not_includes @resolved_scope, @app_two_rat_permission
-        assert_not_includes @resolved_scope, @app_two_bat_permission
-      end
+        should "not contain any permissions for apps with non-delegatbale signin permission the #{publishing_manager_role} has access to" do
```
Comment (Gweaton) [inline test/policies/supported_permission_policy_scope_test.rb:79]: typo in `delegatable` here

**Review by Gweaton — APPROVED** [review]
I think this makes sense, given my limited knowledge of Signon - feels a lot less complex and the new test coverage gives me confidence things won't break!

File: test/policies/supported_permission_policy_scope_test.rb (hunk 1 above)
Comment (yndajas) [reply test/policies/supported_permission_policy_scope_test.rb:29]: Yeah I just went with it 😅 this suite could definitely have a bit more of a refactor, but not right now!

**Pull request merged**